KeyVaultAuthSample.cs

```diff
@@ -134,7 +134,10 @@ public static void Main(String[] args)
         {
             Console.WriteLine("Azure Key Vault Authentication Sample");
             KeyVaultAuthSample sample = new KeyVaultAuthSample();
-            sample.run().Wait();
+
+            // Any synchronous invocation of asyncronous code (such as KV API calls) should follow the below pattern of calling ConfigureAwait(false) to avoid deadlock
+            // Please see https://blog.stephencleary.com/2012/07/dont-block-on-async-code.html
+            sample.run().ConfigureAwait(false).GetAwaiter().GetResult();
```

Comment:
This should be:

Task.Run( () => sample.RunAsync() ).ConfigureAwait( false ).GetAwaiter( ).GetResult( );

Reply:
Also, the name of each async method should end with 'Async'. (I can't seem to add comments to those lines.) 
Public methods should start with a capital, so 'run' becomes 'RunAsync'

Reply:
updated